csaf_2.0/json_schema/csaf_json_schema.json

```diff
@@ -154,13 +154,127 @@
           ]
         },
-        "cpe": {
-          "$comment": "TODO: This should have a full regular expression to enforce CPE syntax.",
-          "title": "Common Platform Enumeration representation",
-          "description": "The Common Platform Enumeration (CPE) attribute refers to a method for naming platforms external to this specification.",
-          "type": "string",
-          "pattern": "^(cpe:2\\.3:[aho\\*\\-](:(((\\?*|\\*?)([a-zA-Z0-9\\-\\._]|(\\\\[\\\\\\*\\?!\"#\\$%&'\\(\\)\\+,/:;<=>@\\[\\]\\^`\\{\\|\\}~]))+(\\?*|\\*?))|[\\*\\-])){5}(:(([a-zA-Z]{2,3}(-([a-zA-Z]{2}|[0-9]{3}))?)|[\\*\\-]))(:(((\\?*|\\*?)([a-zA-Z0-9\\-\\._]|(\\\\[\\\\\\*\\?!\"#\\$%&'\\(\\)\\+,/:;<=>@\\[\\]\\^`\\{\\|\\}~]))+(\\?*|\\*?))|[\\*\\-])){4})|([c][pP][eE]:/[AHOaho]?(:[A-Za-z0-9\\._\\-~%]*){0,6})$",
-          "minLength": 5
+        "product_identification_helper": {
+          "title": "Helper to identify the product",
+          "description": "Provides a least one method which aids in identifying the product in an asset database.",
+          "type": "object",
+          "minProperties": 1,
+          "properties": { 
+            "cpe": {
+              "$comment": "TODO: This should have a full regular expression to enforce CPE syntax.",
+              "title": "Common Platform Enumeration representation",
+              "description": "The Common Platform Enumeration (CPE) attribute refers to a method for naming platforms external to this specification.",
+              "type": "string",
+              "pattern": "^(cpe:2\\.3:[aho\\*\\-](:(((\\?*|\\*?)([a-zA-Z0-9\\-\\._]|(\\\\[\\\\\\*\\?!\"#\\$%&'\\(\\)\\+,/:;<=>@\\[\\]\\^`\\{\\|\\}~]))+(\\?*|\\*?))|[\\*\\-])){5}(:(([a-zA-Z]{2,3}(-([a-zA-Z]{2}|[0-9]{3}))?)|[\\*\\-]))(:(((\\?*|\\*?)([a-zA-Z0-9\\-\\._]|(\\\\[\\\\\\*\\?!\"#\\$%&'\\(\\)\\+,/:;<=>@\\[\\]\\^`\\{\\|\\}~]))+(\\?*|\\*?))|[\\*\\-])){4})|([c][pP][eE]:/[AHOaho]?(:[A-Za-z0-9\\._\\-~%]*){0,6})$",
+              "minLength": 5
+            },
+            "hashes": {
+              "$comment": "TODO: Array of objects (algorithm, value, optional name)",
+              "title": "List of hashes",
+              "description": "Contains a list of cryptographic hashes usable to identify files.",
+              "type": "array",
+              "minItems": 1,
+              "items": {
+                "title": "Cryptographic hash",
+                "description": "Contains all information to identify a file based on its cryptographic hash value.",
+                "type": "object",
+                "required": [
+                  "algorithm",
+                  "file",
+                  "value"
+                ],
+                "properties": {
+                  "algorithm": {
+                    "title": "Algorithm of the cryptographic hash",
+                    "description": "Contains the name of the cryptographic hash algorithm used to calculate the value.",
+                    "type": "string",
+                    "enum": [
```

Comment:
Actually not. I wanted to prevent the use of insecure hash algos... How often do they need to change? With this version a consumer is able to have a fixed set of strings for each software installed in his asset database and can easily compare them.

Reply:
I opt to go for examples (hashing algorithms we should not prescribe).

Reply:
What about a reasonable default value like "SHA-3"?

Reply:
+1 for a reasonable default.